src/HappyEyeBallsConnectionBuilder.php

```diff
@@ -0,0 +1,304 @@
+<?php
+
+namespace React\Socket;
+
+use React\Dns\Model\Message;
+use React\Dns\Resolver\Resolver;
+use React\EventLoop\LoopInterface;
+use React\EventLoop\TimerInterface;
+use React\Promise;
+use React\Promise\CancellablePromiseInterface;
+use InvalidArgumentException;
+use RuntimeException;
+
+/**
+ * @internal
+ */
+final class HappyEyeBallsConnectionBuilder
```

Comment:
Interesting design decision, this really threw me off guard :grin:

Reply:
Same as above, to prevent memory leaks, the class symbolises a new connection attempt and will disappear once that connection has been established.